src/test/java/hudson/plugins/dry/DryWorkflowTest.java

```diff
@@ -0,0 +1,91 @@
+package hudson.plugins.dry;
+
+import hudson.FilePath;
+import hudson.model.Result;
+import hudson.tasks.Maven;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.junit.*;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * In this test suite we initialize the Job workspaces with a resource (maven-project1.zip) that contains a Maven
+ * project with pmd-cpd configured.
+ */
+public class DryWorkflowTest {
+
+    @ClassRule
+    public static JenkinsRule jenkinsRule = new JenkinsRule();
+
+    private static Maven.MavenInstallation mavenInstallation;
+
+    @BeforeClass
+    public static void init() throws Exception {
+        mavenInstallation = jenkinsRule.configureMaven3();
+    }
+
+    /**
+     * Run a workflow job using {@link DryPublisher} and check for success.
+     */
+    @Test
+    public void dryPublisherWorkflowStep() throws Exception {
+        WorkflowJob job = jenkinsRule.jenkins.createProject(WorkflowJob.class, "dryPublisherWorkflowStep");
+        FilePath workspace = jenkinsRule.jenkins.getWorkspaceFor(job);
+        workspace.unzipFrom(getClass().getResourceAsStream("./maven-project1.zip"));
+        job.setDefinition(new CpsFlowDefinition(""
+                        + "node {\n"
+                        + "  def mvnHome = tool '" + mavenInstallation.getName() + "'\n"
+                        + "  sh \"${mvnHome}/bin/mvn clean install\"\n"
```

Comment:
BTW the tests could be faster and simpler and more reliable by using “canned” result files rather than doing a real Maven build. A little less realistic of course.